### DISCUSSION TITLE
Set an @value string and use it elsewhere

### DISCUSSION THREAD

**Discussion by BelgianNoise** [LinkedSoftwareDependencies/Components.js#98] [Q&A] [answered]
I am trying to use a string variable but I want to define it elsewhere, in this case it is in another config file.
Im not sure how to explain myself :) example code might help:

```
    {
      "@id": "urn:some:id",
      "@value": "some-value"
    }
```

```
    {
      "@type": "SomeRequestHandler",
      "aVariableTypedString": {
        "@id": "urn:some:id"
      }
    }
```

Is this possible? I feel like I have done a similar thing before.
Hope my question is clear!

**Comment by rubensworks** [comment, marked as answer]
You're probably thinking of [the variables functionality](https://componentsjs.readthedocs.io/en/latest/configuration/configurations/variables/), which allows you to set variable values at instantiation-time.
Does this help?

**Reply by BelgianNoise** [reply]
Hmm not quite what I am looking for. I am trying to set the value inside the config file so I can use a different config file for every environement.

**Reply by rubensworks** [reply]
Not sure if that's possible. But it _may_ be possible via one of the following:

```
    {
      "@id": "urn:some:id",
      "value": "some-value"
    }
```

```
    {
      "@id": "urn:some:id",
      "valueRaw": "some-value"
    }
```

If it doesn't work with those two, then there is no way to achieve what you need AFAIK. Best to open a feature request in that case.

**Reply by BelgianNoise** [reply]
> ```
>     {
>       "@id": "urn:some:id",
>       "valueRaw": "some-value"
>     }
> ```

YES this is it. Thanks a lot!

**Reply by ixuz** [reply]
A follow up question.
What's the difference between "value" and "valueRaw"?

**Reply by rubensworks** [reply]
@ixuz Internally, they refer to the same predicate: https://github.com/LinkedSoftwareDependencies/Components.js/blob/master/components/context.jsonld#L239-L245
The only difference is that `valueRaw` is interpreted as a literal (string, number, ...), while `value` is interpreted as an IRI (`@id`).
